edb/pgsql/compiler/relgen.py

```diff
@@ -172,7 +174,6 @@ def get_set_rvar(
                 ir_set=ir_set, stmt=stmt, ctx=subctx)
             subctx.pending_query = subctx.rel = stmt
 
-        path_scope = relctx.get_scope(ir_set, ctx=subctx)
         if path_scope:
```

Comment:
A slightly better fix would be to move this `if` block to before `is_optional`.  `update_scope` reassigns `ctx.scope_tree` to a correct value.

Reply:
Argh, disregard. This breaks the optional wrapper.

Reply:
Yeah, that was the first thing I tried, but moving it ahead of the prepare optional excitingly does not work.